test/github-issues/6640/issue-6640.ts

```diff
@@ -0,0 +1,93 @@
+import "reflect-metadata";
+import {closeTestingConnections, createTestingConnections, reloadTestingDatabases} from "../../utils/test-utils";
+import {Connection} from "../../../src/connection/Connection";
+import {Form} from "./entity/Form";
+import {FormTmpl} from "./entity/FormTmpl";
+import { expect } from "chai";
+
+
+
+describe("github issues > #6640 Cannot ORDER BY json field in query with JOIN and pagination", () => {
+
+    let connections: Connection[];
+    before(async () => connections = await createTestingConnections({
+        entities: [__dirname + "/entity/*{.js,.ts}"],
+        enabledDrivers: ["postgres"]
+    }));
+    beforeEach(() => reloadTestingDatabases(connections));
+    after(() => closeTestingConnections(connections));
+
+    it("should correctly run query with join, order by json field and pagination", () => Promise.all(connections.map(async connection => {
+
+      const formTmpls: FormTmpl[] = [
+        {
+          id: 1,
+          origin: 1,
+          name: "form1",
+          publicSettings: [],
+        },
+        {
+          id: 2,
+          origin: 2,
+          name: "form2",
+          publicSettings: [],
+        },
+      ];
+      await connection.getRepository(FormTmpl).save(formTmpls);
+
+      const repo = connection.getRepository(Form);
+      const form1 = new Form();
+      form1.formTmplId = 1;
+      form1.settings = "setting1";
+      form1.extraSettings = {
+        someNb: 7,
+        someStr: "extra1",
+      };
+      await repo.save(form1);
+
+      const form2 = new Form();
+      form2.formTmplId = 2;
+      form2.settings = "setting2";
+      form2.extraSettings = {
+        someNb: 21,
+        someStr: "extra2",
+      };
+      await repo.save(form2);
+
+      const form3 = new Form();
+      form3.formTmplId = 1;
+      form3.settings = "setting3";
+      form3.extraSettings = {
+        someNb: 5,
+        someStr: "extra3",
+      };
+      await repo.save(form3);
+
+      const form4 = new Form();
+      form4.formTmplId = 1;
+      form4.settings = "setting4";
+      form4.extraSettings = {
+        someNb: 40,
+        someStr: "extra4",
+      };
+      await repo.save(form4);
+
+      const orig = 1;
+      const qb = connection.createQueryBuilder(Form, "form")
+        .innerJoinAndSelect("form.formTmpl", "formTmpls", "formTmpls.origin = :orig", { orig })
+        .orderBy({
+          "form.extraSettings ->'someNb'": {
```

Comment:
What about using the actual JSON notation, `form.extraSettings->'someNb'`?

Reply:
Using the JSON notation (without the space before '->') does not work in other places either. Typeorm will not treat them as separate fields, only the '.' is recognized for it. And this will mess up the entity to table column translation.

But using the space before '->' does the trick in most cases (i.e. other parts of the code).